src/TypeScript.UnitTests/NodeJSLocator/Locators/GlobalPathNodeLocatorTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator.Locators;
+
+namespace SonarLint.VisualStudio.TypeScript.UnitTests.NodeJSLocator.Locators
+{
+    [TestClass]
+    public class GlobalPathNodeLocatorTests
+    {
+        private readonly string programFilesPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "nodejs\\node.exe");
```

Comment:
I wanted originally to put these fields in the product class and make them internal, but Environment.SpecialFolder.ProgramFilesX86 returns an empty string on x86 machines, which is handled separately in the product code via the string.IsNullOrEmpty(x86path).
So if I were to use this full calculation in the product code, it would be a relative path instead of an absolute on x86 machines. It probably doesn't make much difference in practice (only if under the current directory we have a "nodejs\node.exe" as well), but I didn't want to make changes to the nodejstools code so it will be easier to compare in the future.